packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,170 @@
+        entityId: sourceEntityId,
+      }),
+      linkTypeModel: await LinkTypeModel.get(graphApi, {
+        versionedUri: linkTypeUri,
+      }),
+      targetEntityModel: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: targetEntityId,
+      }),
+    };
+
+    return new LinkModel({
+      accountId,
+      ...fetchedParams,
+    });
+  }
+
+  /**
+   * Get all outgoing links of a source entity.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntityModel the source entity of the link
+   */
+  static async getAllOutgoing(
+    graphApi: GraphApi,
+    { sourceEntityModel }: { sourceEntityModel: EntityModel },
+  ): Promise<LinkModel[]> {
+    const { data: entityLinks } = await graphApi.getEntityLinks(
+      sourceEntityModel.entityId,
+    );
+    return Promise.all(
+      Object.entries(entityLinks)
+        /**
+         * When target entity id is a list the link has many targets.
+         * This is not supported yet
+         * @todo support 1:N links
+         * */
+        .filter(([_, targetEntityId]) => typeof targetEntityId === "string")
+        .map(
+          async ([linkTypeUri, targetEntityId]) =>
+            new LinkModel({
+              /**
+               * @todo figure out what account ID we use here
+               *   https://app.asana.com/0/1202805690238892/1202883599104674/f */
+              accountId: sourceEntityModel.accountId,
+              linkTypeModel: await LinkTypeModel.get(graphApi, {
+                versionedUri: linkTypeUri,
+              }),
+              sourceEntityModel,
+              targetEntityModel: await EntityModel.getLatest(graphApi, {
+                /**
+                 * @todo figure out what account ID we use here
+                 *   https://app.asana.com/0/1202805690238892/1202883599104674/f */
+                accountId: sourceEntityModel.accountId,
+                entityId: targetEntityId,
+              }),
+            }),
+        ),
+    );
+  }
+
+  /**
+   * Get the outgoing link of a source entity given a link type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntityModel the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   */
+  static async getOutgoing(
+    graphApi: GraphApi,
+    {
+      sourceEntityModel,
+      linkTypeModel,
+    }: { sourceEntityModel: EntityModel; linkTypeModel: LinkTypeModel },
+  ): Promise<LinkModel | null> {
+    /**
+     * @todo use structural querying for this client-side fetch
+     *   https://app.asana.com/0/1200211978612931/1202510174412974/f
+     */
+    const links = (
+      await LinkModel.getAllOutgoing(graphApi, {
+        sourceEntityModel,
+      })
+    ).filter(
+      (link) => link.linkTypeModel.schema.$id === linkTypeModel.schema.$id,
+    );
+    /** @todo this may return an array of links whe new support 1:N links. */
```

Comment:
Tiny nit, still a typo
```suggestion
    /** @todo this may return an array of links when we support 1:N links. */
```

Should we add an asana task here too

Reply:
So sorry, I shouldn't have edited in GitHub :smile: 

713e9e57312790c52e5b86d6544b3252d36a9b87